pkg/operator/ceph/cluster/osd/provision_spec.go

```diff
@@ -157,6 +157,8 @@ func (c *Cluster) provisionPodTemplateSpec(osdProps osdProperties, restart v1.Re
 	p := cephv1.GetOSDPlacement(c.spec.Placement)
 	if !osdProps.onPVC() {
 		p.ApplyToPodSpec(&podSpec)
+	} else if osdProps.getPreparePlacement().NodeAffinity != nil {
+		osdProps.getPreparePlacement().ApplyToPodSpec(&podSpec)
```

Comment:
The merging of node affinity was really only done for mons since the mons have some affinity generated by the operator, to be merged with the placement in the CR. We basically don't want/need to have the merging for OSD pod specs. How about this approach?
- Add a new bool param to `ApplyToPodSpec()` that indicates if it should merge the properties. If false, it would skip the logic that merges the node affinity.
- The OSDs would pass `false`. 
- Other daemons would pass `true`.